### ISSUE TITLE
Fonts are leaking memory

### ISSUE THREAD

**Issue by jonathan-dejong** [mpdf/mpdf#160]
### I have this problem/would like to have this functionality

So I've done some extensive testing and the culprit for leaking memory is the dejavu fonts. 

### These are mPDF and PHP versions I am using
Master branch (assumed 7.0?) with PHP version 5.6.21

Here's the memory usage for the first and second iteration (of many) which shows memory usage after 
`$this->_putpages();` on row ~11342 and `$this->_putfonts();` on row ~26306. I've tested it down even more granular and it is indeed the `_putfonts()`function. I have not tested within that yet tho. 
```
//First run
[28-Jun-2016 13:22:13 Europe/London] after putpages:
[28-Jun-2016 13:22:13 Europe/London] Memory used: 24
[28-Jun-2016 13:22:13 Europe/London] after putfonts:
[28-Jun-2016 13:22:13 Europe/London] Memory used: 30

//Second run
[28-Jun-2016 13:22:13 Europe/London] after putpages:
[28-Jun-2016 13:22:13 Europe/London] Memory used: 30
[28-Jun-2016 13:22:13 Europe/London] after putfonts:
[28-Jun-2016 13:22:13 Europe/London] Memory used: 32.25
```

I figured if I can't stop this completely I should at least be able to significantly lower the number by *only* using arial throughout everything, not even loading in dejavu or any other (I know I don't need them). However I've had no luck in this. I've removed all other fonts than the different dejavu-fonts in ttfonts. I've set arial in my stylesheet. I've changed all instances of devaju in `config_lang2fonts`, `config_fonts` and `mpdf.php`. 

Any idea how to proceed? It's really not worth ~1.5 mb memory for each iteration just to get the dejavu font in my case. 

Thanks!

**Comment by cliffordp** [comment]
I didn't investigate memory leaks, but I would like something generic like "Arial for all the things" without the need for any fonts loaded via this library.

config_fonts.php includes this comment:
`$this->backupSubsFont = array('dejavusanscondensed','arialunicodems','sun-exta');	// this will recognise most scripts` (typo in _Extra_)
but then sets the default to
`$this->backupSubsFont = array('dejavusanscondensed', 'freeserif');`

But the `ttfonts` directory doesn't include _Arial Unicode MS_ -- should it?

**Comment by finwe** [comment]
@cliffordp you are off topic in this thread.

Also please note that the Issue tracker serves mainly as a place to report bugs and request new features as noted in [Contributing guidelines](https://raw.githubusercontent.com/mpdf/mpdf/development/.github/CONTRIBUTING.md). Please do not abuse it as a general questions or troubleshooting location.

Use the the [mpdf tag](https://stackoverflow.com/questions/tagged/mpdf) at [Stack Overflow](https://stackoverflow.com/).

**Comment by cliffordp** [comment]
Ah, thanks for the info. I'll know for next time. Sorry about that.

(Sounds like Arial isn't included by default and I'd have to include it manually, per https://stackoverflow.com/a/43346075/893907)

**Comment by finwe** [comment]
You could always use `c` mPDF mode for core fonts only and use core Helvetica font. Then, no external fonts are used. Helvetica is practically the same font as Arial for basic usage.

see https://mpdf.github.io/fonts-languages/choosing-a-configuration-v5-x.html#core-non-embedded-fonts

**Comment by jonathan-dejong** [comment]
Had not seem that setting! Thanks @finwe 

Tho I should note that you'll get arial or helvetica depending on the machine, not just helvetica :)

**Comment by dmitryd** [comment]
I forgot to add: out of memory always happens in `gzuncompress` when fetching fonts from the font cache. The `.cw.dat` file (131K). Original fonts are small, around 40K on the disk.

**Comment by DPlachkov1991** [comment]
@dmitryd did you ever figure out how to fix this?

**Comment by dmitryd** [comment]
@DPlachkov1991 No. We moved pdf generation from the web server to a separate php script, which we run via `shell_exec` in one case. In another case we have to generate many pdfs from product records. There we watch the amount of remaining memory and restart the script when it gets too low.

**Comment by dmitryd** [comment]
@DPlachkov1991 When it generates a single PDF (like for the current page), it does not run out of memory. It stops and all its memory released. So the web server is not affected.

For the cron job - yes, this is how we do it. But I cannot share, it is a customer's project.

**Comment by DPlachkov1991** [comment]
@dmitryd thank you, you have been quite helpful.

 As for the font leaking memory, maybe the creator @finwe  can look into it if he has time. I tried using the "c" core fonts, but they lack Cyrillic, so using "Ariel" takes 1-2 MB memory each pdf

**Comment by finwe** [comment]
I'm afraid I lack capacity and overview about the issue. I'll be glad to receive and review any reasonable PR attempt to address this.
